Formula/llvm.rb

```diff
@@ -95,46 +120,60 @@ class Llvm < Formula
       url "http://llvm.org/git/lldb.git"
     end
 
-    # Polly is --HEAD-only because it requires isl and the version of Polly
-    # shipped with 3.6.2 only compiles with isl 0.14 and earlier (current
-    # version is 0.15). isl is distributed with the Polly source code from LLVM
-    # 3.7 and up, so --HEAD builds do not need to depend on homebrew isl.
-    option "with-polly", "Build with the experimental Polly optimizer"
     resource "polly" do
       url "http://llvm.org/git/polly.git"
     end
   end
 
   keg_only :provided_by_osx
 
+  # Options
   option :universal
-  option "with-clang", "Build the Clang compiler and support libraries"
+  option "without-clang", "Do not build the Clang compiler and support libraries"
   option "with-clang-extra-tools", "Build extra tools for Clang"
   option "with-compiler-rt", "Build Clang runtime support libraries for code sanitizers, builtins, and profiling"
-  option "with-libcxx", "Build the libc++ standard library"
+  option "without-libcxx", "Do not build the libc++ standard library"
+  option "with-libcxxabi", "Build the libc++abi standard library"
+  option "with-libunwind", "Build the libunwind library"
   option "with-lld", "Build LLD linker"
   option "with-lldb", "Build LLDB debugger"
   option "with-python", "Build Python bindings against Homebrew Python"
-  option "with-rtti", "Build with C++ RTTI"
+  option "without-rtti", "Do not build C++ RTTI"
   option "with-utils", "Install utility binaries"
-
-  deprecated_option "rtti" => "with-rtti"
-
-  if MacOS.version <= :snow_leopard
+  option "with-polly", "Build with the experimental Polly optimizer"
+  option "with-test", "Build LLVM unit tests"
+  option "with-shared-libs", "Build all libs as shared instead of static"
```

Comment:
I think it's best we do not offer this option at all. The LLVM documentation is pretty clear that this is only to be used by developers (people that are working on LLVM itself) and not by users (people that develop with LLVM). The former group is very unlikely to install LLVM via Homebrew.